File: objc/Builder/Builder/Manufacturer.m
#import "Manufacturer.h"

@implementation Manufacturer

-(Computer*)getComputer {
    return [_computerBuilder getComputer];
}

-(void)constructComputer {
    [_computerBuilder createComputer];
    [_computerBuilder buildCPU];
    [_computerBuilder buildMotherboard];
    [_computerBuilder buildRAM];
    [_computerBuilder buildStorage];
    [_computerBuilder buildGraphicsCard];
}

@end

File: swift/Builder/Builder/Manufacturer.swift
import Foundation

public class Manufacturer {
    
    var computerBuilder: ComputerBuilder?
    
    func getComputer() -> Computer {
        return computerBuilder!.getComputer()
    }
    
    func constructComputer() -> Void {
        computerBuilder!.createComputer()
        computerBuilder!.buildCPU()
        computerBuilder!.buildMotherboard()
        computerBuilder!.buildRAM()
        computerBuilder!.buildStorage()
        computerBuilder!.buildGraphicsCard()
    }
}
